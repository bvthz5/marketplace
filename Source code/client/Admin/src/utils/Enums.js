export const orderStatus = {
  CREATED: 0,
  FAILED: 1,
  CONFIRMED: 2,
  WAITING_FOR_PICKUP: 3,
  INTRANSIT: 4,
  OUTFORDELIVERY: 5,
  CANCELLED: 6,
  DELIVERED: 7,
};
