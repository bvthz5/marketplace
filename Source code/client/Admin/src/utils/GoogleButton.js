import React from 'react';
import { GoogleLogin } from '@react-oauth/google';

export const GoogleButton = ({ response }) => {
  return (
    <>
     
        <GoogleLogin onSuccess={response} onError={console.log('Login Failed')} useOneTap />
   
    </>
  );
};
