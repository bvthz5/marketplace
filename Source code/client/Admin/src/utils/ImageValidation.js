import { toast } from "react-toastify";

export const validateProfilePicture = (profilePic) => {
    let allowedExtension = ['jpeg', 'jpg', 'png', 'webp'];
    let fileExtension = profilePic.name.split('.').pop().toLowerCase();
    let isValidFile = false;

    for (let index in allowedExtension) {
      if (fileExtension === allowedExtension[index]) {
        isValidFile = true;
        break;
      }
    }
    if (!isValidFile) {
      toast.error('Allowed Extensions are : *.' + allowedExtension.join(', *.'));
    }
    return isValidFile;
  };