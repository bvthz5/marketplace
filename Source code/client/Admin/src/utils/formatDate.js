export function convertDate(dates) {
  let date = new Date(dates);
  if (isNaN(date.getTime())) {
    return "";
  }
  let month = ['JAN', 'FEB', 'MAR', 'APR', 'MAY', 'JUN', 'JUL', 'AUG', 'SEP', 'OCT', 'NOV', 'DEC'];
  return date.getDate() + ' ' + month[date.getMonth()] + ' ' + date.getFullYear().toString();
}
export  function convertDatesmall(dates) {
  let date = new Date(dates);
  if (isNaN(date.getTime())) {
    return "";
  }
  let month = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
  return date.getDate() + ' ' + month[date.getMonth()] + ' ' + date.getFullYear().toString();
}


export  const getRelativeDate=(date)=> {
  const today = new Date();
  const inputDate = new Date(date);
  const oneDay = 24 * 60 * 60 * 1000; // hours*minutes*seconds*milliseconds
  const diffDays = Math.round(Math.abs((today - inputDate) / oneDay));

  if (diffDays === 0) {
    return 'TODAY';
  } else if (diffDays === 1) {
    return 'YESTERDAY';
  } else if (diffDays < 7) {
    return `${diffDays} DAYS AGO`;
  } else {
    const monthNames = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
    const month = monthNames[inputDate.getMonth()];
    const day = inputDate.getDate();
    return `${month.toUpperCase()} ${day}`;
  }
}
