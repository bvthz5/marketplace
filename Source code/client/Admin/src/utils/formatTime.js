import { format,  formatDistanceToNow } from 'date-fns';

// ----------------------------------------------------------------------

export function fDate(date, newFormat) {
  const fm = newFormat || 'dd MMM yyyy';

  return date ? format(new Date(date), fm) : '';
}

export function fToNow(date) {
  return date
    ? formatDistanceToNow(new Date(date), {
        addSuffix: true,
      })
    : '';
}
