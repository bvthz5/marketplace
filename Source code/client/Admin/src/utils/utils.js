import React from 'react';
import style from '../pages/Product-List-Admin/AdminProductDetailView/AdminProductDetailView.module.css';
import productliststyle from '../pages/Product-List-Admin/ProductList.module.css';
import deliveryagentcss from '../pages/Delivery-Agent/DeliveryAgent.module.css';
import { orderStatus } from './Enums';

export const handleStatus = (status) => {
  if (status === 0) {
    return <span className={style['logged-rejected']}>Rejected</span>;
  }
  if (status === 1) {
    return <span className={style['logged-in']}>Approved</span>;
  }
  if (status === 2) {
    return <span className={style['warning']}>Pending</span>;
  }
  if (status === 3) {
    return <span className={style['logged-sold']}>Sold</span>;
  }
  if (status === 4) {
    return <span className={style['logged-out']}>Deleted</span>;
  }
  if (status === 6) {
    return <span className={style['warning']}>Order Processing</span>;
  }
};

export const handleItemStatus = (status) => {
  switch (status) {
    case orderStatus.CREATED:
      return 'Order created';

    case orderStatus.CONFIRMED:
      return 'Order Confirmed';

    case orderStatus.INTRANSIT:
      return 'In Transit';

    case orderStatus.CANCELLED:
      return 'Order Cancelled';

    case orderStatus.WAITING_FOR_PICKUP:
      return 'Waiting For Pickup';

    case orderStatus.OUTFORDELIVERY:
      return 'Out For Delivery';

    case orderStatus.DELIVERED:
      return 'Delivered';

    default:
      return '';
  }
};

export const PASSWORD_PATTERN = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&#])[A-Za-z\d@$!%*?#&]{8,16}$/;

export const handleProductStatus = (status) => {
  if (status === 0) {
    return <span className={productliststyle['logged-rejected']}>Rejected</span>;
  }
  if (status === 1) {
    return <span className={productliststyle['logged-in']}>Approved</span>;
  }
  if (status === 2) {
    return <span className={productliststyle['logged-pending']}>Pending</span>;
  }
  if (status === 3) {
    return <span className={productliststyle['logged-sold']}>Sold</span>;
  }
  if (status === 4) {
    return <span className={productliststyle['logged-out']}>Deleted</span>;
  }
  if (status === 6) {
    return <span className={productliststyle['logged-orderprocessing']}>Order Processing</span>;
  }
};

export const statusBadge = (status) => {
  switch (status) {
    case 3:
      return (
        <div
          className={deliveryagentcss['statuscss']}
          style={{ color: 'red', backgroundColor: 'rgba(255, 86, 48, 0.16)' }}
        >
          Deleted
        </div>
      );
    case 2:
      return (
        <div className={deliveryagentcss['statuscss']} style={{ color: 'orange', backgroundColor: '#e9981629' }}>
          Blocked
        </div>
      );
    case 1:
      return (
        <div
          className={deliveryagentcss['statuscss']}
          style={{ color: 'green', backgroundColor: 'rgba(7, 92, 49, 0.16)' }}
        >
          Active
        </div>
      );
    case 0:
    default:
      return (
        <div className={deliveryagentcss['statuscss']} style={{ color: '#4a63ee', backgroundColor: '#0f1ef129' }}>
          Inactive
        </div>
      );
  }
};

export function convertTime(time) {
  const date = new Date(time);
  const hours = date.getHours();
  const minutes = date.getMinutes();
  let period = "AM";

  let hour = hours;
  if (hour >= 12) {
    period = "PM";
    hour = hour === 12 ? 12 : hour - 12;
  } else if (hour === 0) {
    hour = 12;
  }

  return `${hour}:${formatTwoDigits(minutes)} ${period}`;
}

function formatTwoDigits(value) {
  return value.toString().padStart(2, "0");
}