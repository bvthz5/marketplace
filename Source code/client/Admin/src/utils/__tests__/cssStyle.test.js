import { bgBlur, bgGradient, textGradient, filterStyles, hideScrollbarY, hideScrollbarX } from '../../utils/cssStyles';

describe('styles', () => {
  describe('bgBlur', () => {
    test('returns expected styles', () => {
      const props = {
        color: '#ff0000',
        blur: 10,
        opacity: 0.5,
        imgUrl: 'https://example.com/image.jpg',
      };
      const styles = bgBlur(props);
      expect(styles).toMatchSnapshot();
    });
  });

  describe('bgGradient', () => {
    test('returns expected styles', () => {
      const props = {
        direction: 'to top',
        startColor: '#ff0000',
        endColor: '#00ff00',
        imgUrl: 'https://example.com/image.jpg',
        color: '#0000ff',
      };
      const styles = bgGradient(props);
      expect(styles).toMatchSnapshot();
    });
  });

  describe('textGradient', () => {
    test('returns expected styles', () => {
      const value = 'red, blue';
      const styles = textGradient(value);
      expect(styles).toMatchSnapshot();
    });
  });

  describe('filterStyles', () => {
    test('returns expected styles', () => {
      const value = 'blur(10px) grayscale(50%)';
      const styles = filterStyles(value);
      expect(styles).toMatchSnapshot();
    });
  });

  describe('hideScrollbarY', () => {
    test('returns expected styles', () => {
      const styles = hideScrollbarY;
      expect(styles).toMatchSnapshot();
    });
  });

  describe('hideScrollbarX', () => {
    test('returns expected styles', () => {
      const styles = hideScrollbarX;
      expect(styles).toMatchSnapshot();
    });
  });
});
