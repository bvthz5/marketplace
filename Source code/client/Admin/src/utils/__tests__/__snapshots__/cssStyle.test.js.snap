// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles bgBlur returns expected styles 1`] = `
Object {
  "&:before": Object {
    "WebkitBackdropFilter": "blur(10px)",
    "backdropFilter": "blur(10px)",
    "backgroundColor": "rgba(255, 0, 0, 0.5)",
    "content": "\\"\\"",
    "height": "100%",
    "left": 0,
    "position": "absolute",
    "top": 0,
    "width": "100%",
    "zIndex": 9,
  },
  "backgroundImage": "url(https://example.com/image.jpg)",
  "position": "relative",
}
`;

exports[`styles bgGradient returns expected styles 1`] = `
Object {
  "background": "linear-gradient(to top, #ff0000, #00ff00), url(https://example.com/image.jpg)",
  "backgroundPosition": "center center",
  "backgroundRepeat": "no-repeat",
  "backgroundSize": "cover",
}
`;

exports[`styles filterStyles returns expected styles 1`] = `
Object {
  "MozFilter": "blur(10px) grayscale(50%)",
  "WebkitFilter": "blur(10px) grayscale(50%)",
  "filter": "blur(10px) grayscale(50%)",
}
`;

exports[`styles hideScrollbarX returns expected styles 1`] = `
Object {
  "&::-webkit-scrollbar": Object {
    "display": "none",
  },
  "msOverflowStyle": "none",
  "overflowX": "scroll",
  "scrollbarWidth": "none",
}
`;

exports[`styles hideScrollbarY returns expected styles 1`] = `
Object {
  "&::-webkit-scrollbar": Object {
    "display": "none",
  },
  "msOverflowStyle": "none",
  "overflowY": "scroll",
  "scrollbarWidth": "none",
}
`;

exports[`styles textGradient returns expected styles 1`] = `
Object {
  "WebkitBackgroundClip": "text",
  "WebkitTextFillColor": "transparent",
  "background": "-webkit-linear-gradient(red, blue)",
}
`;
